### PR TITLE
Fix Crawler failures due to BRE

### PR DESCRIPTION
**Pull request by labkey-ankurj**
#### Rationale
This catches BadRequestExceptions thrown in project views by the templates.

#### Related Pull Requests
https://github.com/LabKey/platform/pull/1493

### CODE REVIEW

File: core/src/org/labkey/core/view/template/bootstrap/PageTemplate.java [hunk 1]
```diff
@@ -291,6 +292,11 @@ public String getProjectTitle()
                     }
                     catch (Exception x)
                     {
+                        if (x instanceof BadRequestException)
+                        {
+                            // re-throw BadRequestException
+                            throw new BadRequestException(x.getMessage(), x);
```
Comment (labkey-ankurj) [inline core/src/org/labkey/core/view/template/bootstrap/PageTemplate.java]: @labkey-jeckels let me know if there is a better approach.

Comment (labkey-jeckels) [reply core/src/org/labkey/core/view/template/bootstrap/PageTemplate.java]: See the changes I just pushed.

File: core/src/org/labkey/core/portal/ProjectController.java [hunk 2]
```diff
@@ -343,9 +345,18 @@ public ModelAndView getView(PageForm form, BindException errors)
...
 
             Template t = isPrint() ? Template.Print : Template.Home;
-            HttpView template = t.getTemplate(getViewContext(), new VBox(), page);
+            try
+            {
+                template = t.getTemplate(getViewContext(), new VBox(), page);
+            }
+            catch (BadRequestException e)
+            {
+                ExceptionUtil.handleException(getViewContext().getRequest(), getViewContext().getResponse(), e, e.getMessage(), false);
```
Comment (labkey-jeckels) [inline core/src/org/labkey/core/portal/ProjectController.java]: Why is this required? I'd expect it to get routed to ExceptionUtil via the standard exception handling, like in other actions, without needing this code

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]

File: core/src/org/labkey/core/portal/ProjectController.java (hunk 2 above)
Comment (labkey-ankurj) [reply core/src/org/labkey/core/portal/ProjectController.java]: It wasn't reaching there before when I kept the debug point there or may be I am wrong. I can re-try without having this code.

Comment (labkey-ankurj) [reply core/src/org/labkey/core/portal/ProjectController.java]: or I guess before the changes to PageTemplate, we were catching the exception in PageTemplate and logging the error but now we are re-throwing so it would reach there.

**Review by labkey-jeckels — APPROVED** [review]
